### PR TITLE
feat: support saving k8s deployment info to upstream

### PR DESCRIPTION
**Pull request by nic-chen**
### Summary
support saving k8s deployment info to upstream, then controller can watch k8s and update the upstream's nodes info.

### CODE REVIEW

**Comment by moonming** [comment]
Is this PR only add schema?

**Comment by nic-chen** [comment]
> Is this PR only add schema?

yep, we just save the info here, other things can be left to the client who invoke api and k8s controller

File: t/admin/routes.t [hunk 1]
```diff
@@ -434,7 +434,7 @@ GET /t
             local code, body = t('/apisix/admin/routes/1',
                  ngx.HTTP_PUT,
                  [[{
-                        "service_id": "invalid_id",
+                        "service_id": "invalid_id$",
```
Comment (moonming) [inline t/admin/routes.t:437]: I think you need a better invalid id, for example `!@#`

File: t/node/upstream.t [hunk 2]
```diff
@@ -26,7 +26,69 @@ run_tests();
 
 __DATA__
 
-=== TEST 1: set upstream(id: 1)
+=== TEST 1: set upstream(id: 1) invalid parameters
```
Comment (membphis) [inline t/node/upstream.t:29]: It is best to add new test cases to the end of the file. This will not affect any existing test cases.
@nic-chen

**Pull request merged**